### DISCUSSION TITLE
对paopaogw clash规则的疑问

### DISCUSSION THREAD

**Discussion by Lovest20018** [kkkgo/PaoPaoGateWay#32] [问和答]
首先非常感谢kkkgo提供了paopaodns跟paopaogw整套解决方案，刚好能解决我的痛点。

我这边现在有几个疑问想请教下，
1.因为已经通过ppdns做了分流了，那对于ppgw来说clash规则我还用现在的metaclash的规则：

`proxy-groups:
  - {name: 🚀 节点选择, type: select, proxies: [🌏 所有节点, 🇭🇰 香港节点, 🇹🇼 台湾节点, 🇯🇵 日本节点, 🇰🇷 韩国节点, 🇸🇬 新加坡节点, 🇺🇸 美国节点, 🇺🇳 其他节点]}

  - {name: 🎯 国内流量, type: select, proxies: [DIRECT, 🚀 节点选择]}
  
  - {name: 🏠 私有网络, type: select, proxies: [🎯 国内流量, 🚀 节点选择]}

  - {name: 🛑 广告拦截, type: select, proxies: [REJECT, 🎯 国内流量, 🚀 节点选择]}

  - {name: 📹 YouTube, type: select, proxies: [🚀 节点选择, 🇭🇰 香港节点, 🇹🇼 台湾节点, 🇯🇵 日本节点, 🇰🇷 韩国节点, 🇸🇬 新加坡节点, 🇺🇸 美国节点, 🇺🇳 其他节点]}

  - {name: 📺 Netflix, type: select, proxies: [🚀 节点选择, 🇭🇰 香港节点, 🇹🇼 台湾节点, 🇯🇵 日本节点, 🇰🇷 韩国节点, 🇸🇬 新加坡节点, 🇺🇸 美国节点, 🇺🇳 其他节点]}

  - {name: 🧬 MisakaF, type: select, proxies: [🎯 国内流量, 🚀 节点选择, 🇭🇰 香港节点, 🇹🇼 台湾节点, 🇯🇵 日本节点, 🇰🇷 韩国节点, 🇸🇬 新加坡节点, 🇺🇸 美国节点, 🇺🇳 其他节点]}

  - {name: 🎞 Emby, type: select, proxies: [🚀 节点选择, 🎯 国内流量, 🇭🇰 香港节点, 🇹🇼 台湾节点, 🇯🇵 日本节点, 🇰🇷 韩国节点, 🇸🇬 新加坡节点, 🇺🇸 美国节点, 🇺🇳 其他节点]}

  - {name: 🧸 PikPak, type: select, proxies: [🚀 节点选择, 🎯 国内流量, 🇭🇰 香港节点, 🇹🇼 台湾节点, 🇯🇵 日本节点, 🇰🇷 韩国节点, 🇸🇬 新加坡节点, 🇺🇸 美国节点, 🇺🇳 其他节点]}

  - {name: 📈 网络测试, type: select, proxies: [🎯 国内流量, 🚀 节点选择, 🇭🇰 香港节点, 🇹🇼 台湾节点, 🇯🇵 日本节点, 🇰🇷 韩国节点, 🇸🇬 新加坡节点, 🇺🇸 美国节点, 🇺🇳 其他节点]}

  - {name: 📢 谷歌服务, type: select, proxies: [🚀 节点选择, 🎯 国内流量, 🇭🇰 香港节点, 🇹🇼 台湾节点, 🇯🇵 日本节点, 🇰🇷 韩国节点, 🇸🇬 新加坡节点, 🇺🇸 美国节点, 🇺🇳 其他节点]}

  - {name: 🎬 国外媒体, type: select, proxies: [🚀 节点选择, 🎯 国内流量, 🇭🇰 香港节点, 🇹🇼 台湾节点, 🇯🇵 日本节点, 🇰🇷 韩国节点, 🇸🇬 新加坡节点, 🇺🇸 美国节点, 🇺🇳 其他节点]}

  - {name: 📲 电报消息, type: select, proxies: [🚀 节点选择, 🇭🇰 香港节点, 🇹🇼 台湾节点, 🇯🇵 日本节点, 🇰🇷 韩国节点, 🇸🇬 新加坡节点, 🇺🇸 美国节点, 🇺🇳 其他节点]}
  
  - {name: 🎮 游戏平台, type: select, proxies: [🎯 国内流量, 🚀 节点选择, 🇭🇰 香港节点, 🇹🇼 台湾节点, 🇯🇵 日本节点, 🇰🇷 韩国节点, 🇸🇬 新加坡节点, 🇺🇸 美国节点, 🇺🇳 其他节点]}
  
  - {name: Ⓜ️ 微软服务, type: select, proxies: [🎯 国内流量, 🚀 节点选择, 🇭🇰 香港节点, 🇹🇼 台湾节点, 🇯🇵 日本节点, 🇰🇷 韩国节点, 🇸🇬 新加坡节点, 🇺🇸 美国节点, 🇺🇳 其他节点]}
  
  - {name: 🍎 苹果服务, type: select, proxies: [🎯 国内流量, 🚀 节点选择, 🇭🇰 香港节点, 🇹🇼 台湾节点, 🇯🇵 日本节点, 🇰🇷 韩国节点, 🇸🇬 新加坡节点, 🇺🇸 美国节点, 🇺🇳 其他节点]}

  - {name: 👓 国内媒体, type: select, proxies: [🎯 国内流量, 🚀 节点选择, 🇭🇰 香港节点, 🇹🇼 台湾节点, 🇯🇵 日本节点, 🇰🇷 韩国节点, 🇸🇬 新加坡节点, 🇺🇸 美国节点, 🇺🇳 其他节点]}

  - {name: 🐟 漏网之鱼, type: select, proxies: [🚀 节点选择, 🎯 国内流量, 🇭🇰 香港节点, 🇹🇼 台湾节点, 🇯🇵 日本节点, 🇰🇷 韩国节点, 🇸🇬 新加坡节点, 🇺🇸 美国节点, 🇺🇳 其他节点]}
  

rule-providers:
  Inside:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/Lovest20018/clash-rules/main/Inside.yaml"
    path: ./ruleset/Inside.yaml
    interval: 86400

  lan:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/blackmatrix7/ios_rule_script/master/rule/Clash/Lan/Lan.yaml"
    path: ./ruleset/lan.yaml
    interval: 86400

  ads:
    type: http
    behavior: domain
    url: "https://raw.githubusercontent.com/privacy-protection-tools/anti-AD/master/anti-ad-clash.yaml"
    path: ./ruleset/ads.yaml
    interval: 86400

  google:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/blackmatrix7/ios_rule_script/master/rule/Clash/Google/Google.yaml"
    path: ./ruleset/google.yaml
    interval: 86400

  microsoft:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/blackmatrix7/ios_rule_script/master/rule/Clash/Microsoft/Microsoft.yaml"
    path: ./ruleset/microsoft.yaml
    interval: 86400

  apple:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/blackmatrix7/ios_rule_script/master/rule/Clash/Apple/Apple.yaml"
    path: ./ruleset/apple.yaml
    interval: 86400

  telegram:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/blackmatrix7/ios_rule_script/master/rule/Clash/Telegram/Telegram.yaml"
    path: ./ruleset/telegram.yaml
    interval: 86400

  game:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/blackmatrix7/ios_rule_script/master/rule/Clash/Game/Game.yaml"
    path: ./ruleset/game.yaml
    interval: 86400 

  youtube:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/blackmatrix7/ios_rule_script/master/rule/Clash/YouTube/YouTube.yaml"
    path: ./ruleset/youtube.yaml
    interval: 86400 

  netflix:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/blackmatrix7/ios_rule_script/master/rule/Clash/Netflix/Netflix.yaml"
    path: ./ruleset/netflix.yaml
    interval: 86400 

  misakaf:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/Lovest20018/clash-rules/main/MisakaF.yaml"
    path: ./ruleset/misakaf.yaml
    interval: 86400

  emby:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/Lovest20018/clash-rules/main/Emby.yaml"
    path: ./ruleset/emby.yaml
    interval: 86400

  pikpak:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/blackmatrix7/ios_rule_script/master/rule/Clash/PikPak/PikPak.yaml"
    path: ./ruleset/pikpak.yaml
    interval: 86400

  networktest:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/blackmatrix7/ios_rule_script/master/rule/Clash/Speedtest/Speedtest.yaml"
    path: ./ruleset/networktest.yaml
    interval: 86400

  globalmedia:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/blackmatrix7/ios_rule_script/master/rule/Clash/GlobalMedia/GlobalMedia.yaml"
    path: ./ruleset/globalmedia.yaml
    interval: 86400

  chinamedia:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/blackmatrix7/ios_rule_script/master/rule/Clash/ChinaMedia/ChinaMedia.yaml"
    path: ./ruleset/chinamedia.yaml
    interval: 86400

  china:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/blackmatrix7/ios_rule_script/master/rule/Clash/China/China_Classical.yaml"
    path: ./ruleset/china.yaml
    interval: 86400

  chinaips:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/blackmatrix7/ios_rule_script/master/rule/Clash/ChinaIPs/ChinaIPs_Classical_No_IPv6.yaml"
    path: ./ruleset/chinaips.yaml
    interval: 86400

rules:
  - RULE-SET,lan,🏠 私有网络,no-resolve
  - RULE-SET,Inside,🏠 私有网络,
  - RULE-SET,ads,🛑 广告拦截
  - RULE-SET,youtube,📹 YouTube
  - RULE-SET,netflix,📺 Netflix
  - RULE-SET,misakaf,🧬 MisakaF
  - RULE-SET,emby,🎞 Emby
  - RULE-SET,pikpak,🧸 PikPak
  - RULE-SET,networktest,📈 网络测试
  - RULE-SET,google,📢 谷歌服务
  - RULE-SET,globalmedia,🎬 国外媒体
  - RULE-SET,telegram,📲 电报消息
  - RULE-SET,game,🎮 游戏平台
  - RULE-SET,microsoft,Ⓜ️ 微软服务
  - RULE-SET,apple,🍎 苹果服务
  - RULE-SET,chinamedia,👓 国内媒体
  - RULE-SET,china,🎯 国内流量
  - RULE-SET,chinaips,🎯 国内流量
  - MATCH,🐟 漏网之鱼`

因为现在已经在ppdns分流了，那这里的是国内的域名跟ip部分的的rules就完全可以去掉了吧？没任何意义了。因为都不会到ppgw了吧？

2.对于微软，苹果这些国内有服务器的，那分流规则是否也没效了呢？假如要让他们强制过ppgw，应该在ppdns的force_forward_list填上他们的所有域名，让他们强制转发到ppgw吗？假如我设置了force_forward_list，而规则这里又选择了DIRECT，这样是否有问题呢？

3.对于规则里去广告部分的规则，是否还有存在的必要呢？这里应该只是过滤过ppgw的，
而对于ppdns的话-上面套一个adgh的情况下，应该是指定adgh的上游为ppdns了吧？假如套了adgh的话，clash规则里还是否有需要广告拦截规则存在呢？

**Comment by kkkgo** [comment]
>1. 国内的域名跟ip部分的的rules就完全可以去掉了吧？没任何意义了。因为都不会到ppgw了吧  

确实，如果你是静态路由+FAKEIP网关的配置的话，国内域名不解析出FAKEIP的网段就不会走到ppgw，除非你手动把域名加到ppdns或者在主路由加国内IP的静态路由才会走到ppgw。

>2. 对于微软，苹果这些国内有服务器的，那分流规则是否也没效了呢

很难说，取决于你里面的规则怎么写。比如预置的`force_nocn_list.txt`和`force_forward_list.txt`有部分推荐规则，因为bing有全球版所以把bing.com forward，因为苹果推送不用走代理所以把push.apple.com nocn。在你开了AUTO_FORWARD的话，访问国内没有的服务应该是境外IP，因此会走到ppgw。所以如果你的规则是仅限于境外的微软和苹果域名，那他是生效的。`假如要让他们强制过ppgw，应该在ppdns的force_forward_list填上他们的所有域名，让他们强制转发到ppgw吗` 可以。`假如我设置了force_forward_list，而规则这里又选择了DIRECT`那样就相当于脱裤子放屁，都DIRECT了forward过去跟不forward有啥区别。

>3.对于规则里去广告部分的规则，是否还有存在的必要呢？这里应该只是过滤过ppgw的，
而对于ppdns的话-上面套一个adgh的情况下，应该是指定adgh的上游为ppdns了吧？假如套了adgh的话，clash规则里还是否有需要广告拦截规则存在呢？

都是基于域名过滤广告的话，套了adgh自然没必要再套一层浪费性能。


最后，最简单也就最稳定的方式就是开fast_node=yes，不用任何规则，如果你不需要不同节点访问不同网站的话。

**Comment by Lovest20018** [comment]
非常感谢回复答疑。
分流对我来说有时候还是需要，毕竟有2个机场，nf可能需要用新加坡，而主要用hk。
对于第三点我还有些疑惑的是，套了adgh的话，最终所有的流量还是会过adgh一遍吗？所以去广告加这里是有效果的？
虽然adgh是作为ppdns的下游？

**Reply by kkkgo** [reply]
adgh过滤广告就是把广告域名不返回正确的解析结果，你客户端设置了adgh为DNS，不管adgh的上游是什么，他在请求上游之前就已经拦截了。

**Comment by Lovest20018** [comment]
感谢大佬的指点，挂上adg了。
现在又遇到netflix在tv端上会提示在使用代理。节点的解锁应该是没问题的，毕竟我在用奶昔。
ppgw也换了前置嗅探的，还是一样，换原版clash meta 开启Sniffer ，面板上没有看到netflix的ip被嗅探出来。
是否需要把netflxi的ip全部加到force_forward_list.txt上呢？

**Reply by kkkgo** [reply]
force_forward_list是用来加域名的，加IP去你的主路由加静态路由。
你把IPV6关了试试，不分配IPV6地址看看还提示代理不。

**Reply by kkkgo** [reply]
参考 https://github.com/kkkgo/PaoPaoDNS/discussions/41

**Reply by kkkgo** [reply]
或者参考这个，在主路由上阻止8.8.8.8和8.8.4.4这两个IP。https://vpnreviewer.com/androidtv-streaming-unblocking

**Reply by Lovest20018** [reply]
感谢-我关了ipv6还是一样，我试着组织下这两个ip。额，tp下面要怎么组织ip访问，，我看下里面的说明。

**Comment by Lovest20018** [comment]
tp的5408M说是企业版固件-貌似并没有找到防火墙设置，我设置静态路由-把8.8.8.8 下一跳到0.0.0.0貌似不行。。

**Comment by kkkgo** [comment]
你是什么客户端，浏览器还是APP，浏览器应该不会出问题。
可以把终端设备的DNS和网关都设置成ppgw，开net_rec功能抓一下流量，看是怎么检测的。

**Comment by Lovest20018** [comment]
我用的shiled tv盒子，手动设置了网关跟dns都是ppgw的话也是这样的提示。
手机端跟网页端都是正常的。安卓tv端就正常。


[data.csv](https://github.com/kkkgo/PaoPaoGateWay/files/13227077/data.csv)

**Reply by kkkgo** [reply]
如果手动设置终端的网关和DNS都是ppgw按道理他是走全局ppgw的，我怀疑是你的分流规则可能有问题，你ppgw设置成Global不分流看看。

**Comment by Lovest20018** [comment]
测试了下，clash用的global的话，dns是clash可以播放了。但dns换回ppdns的话还是提示用代理。
这样应该不是规则的问题吧？

**Reply by kkkgo** [reply]
你网关设置的是ppgw的也会提示用代理？

**Comment by Lovest20018** [comment]
嗯。只要dns不是ppgw都是会提示代理。dns跟网关都设置为ppgw的时候抓了下。
[data1.csv](https://github.com/kkkgo/PaoPaoGateWay/files/13227503/data1.csv)
看流量最大的都是nf的新加坡的ip。

**Reply by kkkgo** [reply]
你把DNS设置成8.8.8.8会提示代理吗？

**Comment by Lovest20018** [comment]
dns设置为8.8.8.8 与8.8.4.4 的话不会提示代理，可以正常播放。网关是ppgw

**Reply by kkkgo** [reply]
你把8.8.8.8下一跳随便设置一个内网IP，阻止8.8.8.8访问，再换ppdns试试。

**Comment by Lovest20018** [comment]
![Snipaste_2023-11-01_21-53-44](https://github.com/kkkgo/PaoPaoGateWay/assets/30313973/7b873cde-dc3a-4a5e-a68a-590b598bf93e)
这样设置了，如果dns是ppdns也不行，只有在dns是ppgw或者8.8下面，clash全局下才可以。我看了下nf规则是下面是有很多ip的。

**Reply by kkkgo** [reply]
先不要管规则，IP规则在ppgw里面其实不会起作用。新建一个adg，只转发到ppgw的DNS。然后客户端指定这个adg为DNS。如果可以播放，看看都请求了哪些域名。再看看这些域名用ppdns解析有什么差异。

**Reply by kkkgo** [reply]
另一个测试是你可以把ppdns的AUTO_FORWARD_CHECK设置为no试试。

**Comment by Lovest20018** [comment]
新建的adg上游ppgw的话，是可以播放的。
看了下日志。
C:\Users\com>nslookup ichnaea.netflix.com 192.168.1.252
服务器:  PaoPaoDNS,blog.03k.org
Address:  192.168.1.252

名称:    ichnaea.netflix.com
Address:  198.18.0.16


C:\Users\com>nslookup nrdp.prod.cloud.netflix.com

C:\Users\com>nslookup nrdp.prod.cloud.netflix.com 192.168.1.252
服务器:  PaoPaoDNS,blog.03k.org
Address:  192.168.1.252

名称:    nrdp.prod.cloud.netflix.com
Address:  198.18.0.19


C:\Users\com>nslookup nrdp-ipv6.prod.ftl.netflix.com 192.168.1.252
服务器:  PaoPaoDNS,blog.03k.org
Address:  192.168.1.252

名称:    nrdp-ipv6.prod.ftl.netflix.com
Address:  198.18.0.17


C:\Users\com>nslookup logs.netflix.com 192.168.1.252
服务器:  PaoPaoDNS,blog.03k.org
Address:  192.168.1.252

名称:    logs.netflix.com
Address:  198.18.0.18


C:\Users\com>nslookup occ-0-58-64.1.nflxso.net 192.168.1.252
服务器:  PaoPaoDNS,blog.03k.org
Address:  192.168.1.252

名称:    occ-0-58-64.1.nflxso.net
Address:  198.18.0.23



C:\Users\com>nslookup connectivitycheck.gstatic.com 192.168.1.252
服务器:  PaoPaoDNS,blog.03k.org
Address:  192.168.1.252

非权威应答:
名称:    connectivitycheck.gstatic.com
Address:  203.208.40.34


C:\Users\com>nslookup push.prod.netflix.com 192.168.1.252
服务器:  PaoPaoDNS,blog.03k.org
Address:  192.168.1.252

名称:    push.prod.netflix.com
Address:  198.18.0.72

貌似都被劫持到了fakip地址了。除了connectivitycheck.gstatic.com

**Reply by kkkgo** [reply]
AUTO_FORWARD_CHECK设置为no呢

**Comment by Lovest20018** [comment]
设置过了-单独没用-现在就是在AUTO_FORWARD_CHECK 为no下

**Reply by kkkgo** [reply]
如果设置的DNS解析结果完全一样的话，应该不会出现一个能播放一个不能播放。你可以再对比下DNS请求记录的差异。

**Reply by kkkgo** [reply]
或者把所有请求的域名，都写进forward_list看看。

**Reply by kkkgo** [reply]
另外不知道你有没有注意到以上测试都是要求在设备设置为网关设置为ppgw的IP的前提下，只更换设备DNS不更换网关IP来测试的，如果你是设置ppdns+不是ppgw网关做的测试就无意义了。

**Comment by Lovest20018** [comment]
..突然网络崩了。搞到现在。
我解决了。在v2上找了一个ip地址段，
把下面这些ip也加到静态路由里就行了。
23.246.0.0/18
37.77.184.0/21
45.57.0.0/17
64.120.128.0/17
66.197.128.0/17
108.175.32.0/20
192.173.64.0/18
198.38.96.0/19
198.45.48.0/20

顺便备注下了tg的ip
109.239.140.0/24
149.154.160.0/20
5.28.192.0/18 
91.108.0.0/16

**Reply by kkkgo** [reply]
如果是IP段能解决的话，设置ppgw为网关没理由不能播放，你之前的测试是只设置了ppdns没设置设备的网关为ppgw的IP吧

**Reply by Lovest20018** [reply]
设置ppgw的网关是不能播放，如果dns为ppdns的话，只能设置dns为ppgw的地址或者8.8.8.8下面才能播放。

**Reply by kkkgo** [reply]
没看懂你的描述，`如果dns为ppdns的话，只能设置dns为ppgw的地址`是啥意思
假设10.10.10.3/24是你ppgw的虚拟机IP，10.10.10.8是你ppdns的docker的IP，10.10.10.5是你的设备，那么你的设备应该是设置成这样测试：
IP 10.10.10.5
子网掩码255.255.255.0
网关10.10.10.3
DNS 10.10.10.8

**Reply by Lovest20018** [reply]
IP 10.10.10.5
子网掩码255.255.255.0
网关10.10.10.3
DNS 10.10.10.3
or dns 8.8.8.8
测试的时候我固定ip这样设置就可以了。
现在加了那几条静态路由。dns设置为ppdns也可以了。

**Reply by Lovest20018** [reply]
https://www.v2ex.com/t/824484
看了这个帖子解决的。是否因为这些cdn ip的问题呢？
哦还有我改了netflix的分流规则。之前的规则里没有ip。现在加上了所有ip地址.在静态路由设置后如果这些ip跳到网关，那规则里包含ip是否生效？

**Reply by kkkgo** [reply]
问题是，如果是因为IP直连引起的问题的话，把设备的网关设置为ppgw，那相当于所有IP都走了代理了，跟DNS就没啥关系了。
你试试设置成114.114.114.114看看能不能播？
顺便tg的IP段的话文档有连接（可以点开）。

**Reply by Lovest20018** [reply]
果然，网关设置为ppgw的话，设为国内的都可以播放了。
 那我之前测试的时候应该是设置正常都是有问题的。
 特意把静态路由关了测试了。不可达后立刻就不行了。开着就可以。现在的网络dns是ppdns。网关是主路由的状态下。

 tg文档里的官方地址子网掩码输入后其实会重复吧？

**Reply by kkkgo** [reply]
没有重复啊，tg官方文档的IP段划分的细一些而已，怎么看都不像是拿到了/16这么大的地址。
所以你现在设置ppdns+ppgw的网关其实是可以播放的（跟114一样）？

**Reply by Lovest20018** [reply]
现在网关是主路由也可以播放了，但禁用静态路由就不行。
我刚才网络崩了或许就是因为我网关设置为ppgw,dns也是ppgw。还有什么设置忘记了。直接让pve ip不能访问。2个adg的容器崩溃掉，忙中出错，要重启tp 主路由点到初始化。。。。还好已经备份过。。
太晚了。十分感谢大佬。。
